File: javascript/src/core/entities/district.ts
import { AdministrativeUnit } from "./administrative-unit";

/**
 * District entity.
 */
export class District extends AdministrativeUnit {
  constructor(
    code: string,
    name: string,
    public readonly regencyCode: string
  ) {
    super(code, name);
    if (!regencyCode) throw new Error("Regency code cannot be empty.");
  }
}

File: python/data_wilayah_indonesia/src/data_wilayah_indonesia/core/entities/district.py
from dataclasses import dataclass
from .administrative_unit import AdministrativeUnit

@dataclass(frozen=True)
class District(AdministrativeUnit):
    """
    Entitas untuk tingkat Kecamatan.
    """
    regency_code: str

    def __post_init__(self):
        super().__post_init__()
        if not self.regency_code:
            raise ValueError("Regency code tidak boleh kosong.")
